pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -980,14 +980,14 @@ func validateVSphere(m *Machine, clusterID string) (bool, []string, utilerrors.A
 
 	errs = append(errs, validateVSphereNetwork(providerSpec.Network, field.NewPath("providerSpec", "network"))...)
 
-	if providerSpec.NumCPUs < minVSphereCPU {
-		warnings = append(warnings, fmt.Sprintf("providerSpec.numCPUs: %d is less than the minimum value (%d): the minimum value will be used instead", providerSpec.NumCPUs, minVSphereCPU))
+	if providerSpec.NumCPUs < minVSphereCPU || providerSpec.NumCPUs == 0 {
```

Comment:
Don't think we actually need the or statement here, 0 < minVSphereCPU is true anyway. Similar goes for the cases below